src/metrics/abc.rs

```diff
@@ -0,0 +1,803 @@
+    pub fn conditions_sum(&self) -> f64 {
+        self.conditions_sum
+    }
+
+    /// Returns the `Abc` conditions average value.
+    ///
+    /// This value is computed dividing the `Abc`
+    /// conditions value for the number of spaces.
+    pub fn conditions_average(&self) -> f64 {
+        self.conditions_sum() / self.space_count as f64
+    }
+
+    /// Returns the `Abc` conditions minimum value.
+    pub fn conditions_min(&self) -> f64 {
+        self.conditions_min
+    }
+
+    /// Returns the `Abc` conditions maximum value.
+    pub fn conditions_max(&self) -> f64 {
+        self.conditions_max
+    }
+
+    /// Returns the `Abc` magnitude metric value.
+    pub fn magnitude(&self) -> f64 {
+        (self.assignments.powi(2) + self.branches.powi(2) + self.conditions.powi(2)).sqrt()
+    }
+
+    /// Returns the `Abc` magnitude sum metric value.
+    pub fn magnitude_sum(&self) -> f64 {
+        (self.assignments_sum.powi(2) + self.branches_sum.powi(2) + self.conditions_sum.powi(2))
+            .sqrt()
+    }
+
+    #[inline(always)]
+    pub(crate) fn compute_sum(&mut self) {
+        self.assignments_sum += self.assignments;
+        self.branches_sum += self.branches;
+        self.conditions_sum += self.conditions;
+    }
+
+    #[inline(always)]
+    pub(crate) fn compute_minmax(&mut self) {
+        self.assignments_min = self.assignments_min.min(self.assignments);
+        self.assignments_max = self.assignments_max.max(self.assignments);
+        self.branches_min = self.branches_min.min(self.branches);
+        self.branches_max = self.branches_max.max(self.branches);
+        self.conditions_min = self.conditions_min.min(self.conditions);
+        self.conditions_max = self.conditions_max.max(self.conditions);
+        self.compute_sum();
+    }
+}
+
+#[doc(hidden)]
+pub trait Abc
+where
+    Self: Checker,
+{
+    fn compute(_node: &Node, _stats: &mut Stats) {}
+}
+
+/// Counts Java unary conditional expressions by exploring parenthesis and not operators.
+/// If successful, it returns an Option containing a pair of values representing:
+/// - The unary condition count.
+/// - The presence of a not operator.
+///
+/// Returns None if it fails to explore deeper.
+///
+/// "A unary conditional expression is an implicit condition that uses no relational operators."
+///
+/// - Jerry Fitzpatrick (1997) - "Applying the ABC Metric to C, C++, and Java":
+/// https://www.softwarerenovation.com/Articles.aspx (page 5)
+///
+/// "These are cases where a single variable/field/value is treated as a boolean value.
+/// Examples include if (x) and return !ready."
+///
+/// - GMetrics (a Groovy source code analyzer):
+/// https://dx42.github.io/gmetrics/metrics/AbcMetric.html
+/// - GMetrics test examples:
+/// https://github.com/dx42/gmetrics/blob/master/src/test/groovy/org/gmetrics/metric/abc/AbcMetric_MethodTest.groovy
+fn count_java_unary_conditions(node: &Node) -> Option<(f64, bool)> {
```

Comment:
Why does this function return an `Option` value? It always returns the `Some` at the end because there are `break`s inside the `while let` loop. We can remove it